src/workerd/api/global-scope.c++

```diff
@@ -300,7 +297,7 @@ kj::Promise<DeferredProxy<void>> ServiceWorkerGlobalScope::request(
     return ioContext.awaitJs(promise->then(kj::implicitCast<jsg::Lock&>(lock),
         ioContext.addFunctor(
             [&response, allowWebSocket = headers.isWebSocket(),
-             canceled = kj::addRef(*canceled), &headers]
+             canceled = kj::addRef(*canceled), &headers, &flags]
```

Comment:
`flags` is a local stack variable, so it's unsafe to capture by reference in an async callback.

Since it's a capnp reader and the underlying message has the same lifetime as the isolate, you can capture it by value instead.